vterebey/budgetcalculator/budget_calculator.py

```diff
@@ -0,0 +1,306 @@
+"""
+'Budget Calculator' program to track expenses by category and date.
+To use you need to run the program. Next you need to choose what you
+are going to do, you can:
+1. Enter or select an expense category and cost
+2. View spending statistics on a pie chart
+3. Display expenses by category
+(you can choose the suggested one or enter it yourself)
+4. Display expenses by month (you must enter the year and
+month in the format: YYYY-MM)
+5. Display expenses by category and month at the same time
+(see above how the category and month are set)
+All expenses are recorded in the file 'values.csv'
+The file contains the category,date and cost
+"""
+
+
+import csv
+from datetime import date
+import matplotlib.pyplot as plt
+
+
+# dict 'categories' to store categories and amount
+fieldnames = ["Category", "Date", "Amount"]
+categories = {}
+default_categories = {
+    1: "Grocery store",
+    2: "Clothing",
+    3: "Entertainment",
+    4: "Cafe",
+    5: "Store",
+    6: "Pharmacy",
+}
+with open("values.csv", "w", encoding="UTF-8", newline="") as file:
+    writer = csv.writer(file)
+    writer.writerow(fieldnames)
+
+
+def add_category(category, amount):
+    """
+    adds category and it's amount to the dictionary
+    for displaying statistics
+    """
+    if category not in categories:
+        categories[category] = amount
+    else:
+        categories[category] += amount
+    return categories
+
+
+def get_category(category):
+    """displays expenses depending on the category"""
+    with open("values.csv", "r", encoding="UTF-8", newline="") as csvfile:
+        reader = csv.DictReader(csvfile, delimiter=",")
+        print(*fieldnames, sep=",")
+        for row in reader:
+            if row["Category"] == category:
+                print(f"{row['Category']},{row['Date']},{row['Amount']}")
+        print()
+
+
+def get_month(month):
+    """displays expenses depending on the month"""
+    with open("values.csv", "r", encoding="UTF-8", newline="") as csvfile:
+        reader = csv.DictReader(csvfile, delimiter=",")
+        print(*fieldnames, sep=",")
+        for row in reader:
+            check_month = row["Date"]
+            if check_month[:7] == month:
```

Comment:
why :7 ?
if you want to manipulate dates, then u should use datetime library


```
import dateutil.parser as parser
>>> parser.parse("2020-01-01 14:00")
```


or this is prefarrable variant 

```
import datetime

date_time_str = '2022-12-01 10:27:03.929149'
# strptime(input_string, input_format)
date_time_obj = datetime.datetime.strptime(date_time_str, '%Y-%m-%d %H:%M:%S.%f')

print('Date:', date_time_obj.date())
print('Time:', date_time_obj.time())
print('Date-time:', date_time_obj)
```

Reply:
The second option will be used for date manipulation